Python/bytecodes.c

```diff
@@ -132,24 +132,36 @@ dummy_func(
         inst(NOP, (--)) {
         }
 
+        family(RESUME, 0) = {
+            RESUME_QUICK,
+        };
+
         inst(RESUME, (--)) {
+            TIER_ONE_ONLY
             assert(frame == tstate->current_frame);
-            /* Possibly combine this with eval breaker */
             if (_PyFrame_GetCode(frame)->_co_instrumentation_version != tstate->interp->monitoring_version) {
                 int err = _Py_Instrument(_PyFrame_GetCode(frame), tstate->interp);
                 ERROR_IF(err, error);
-                #if TIER_ONE
                 next_instr--;
-                #endif
-                #if TIER_TWO
-                goto deoptimize;
-                #endif
             }
-            else if (oparg < 2) {
-                CHECK_EVAL_BREAKER();
+            else {
+                if (oparg < 2) {
+                    CHECK_EVAL_BREAKER();
+                }
+                next_instr[-1].op.code = RESUME_QUICK;
             }
         }
 
+        inst(RESUME_QUICK, (--)) {
```

Comment:
Maybe choose a different name? We already have Python versions shipped where `RESUME_QUICK` means "quickened `RESUME` instruction". Perhaps something that indicates it's suitable for higher tiers, such as `RESUME_OR_DEOPT` or `RESUME_OPTIMIZED`? I don't know, naming is hard...

Reply:
I think I'll go with `RESUME_CHECK` as it just checks if anything needs to be done, rather than doing any of the work.